test/test_basebackup.py

```diff
@@ -625,12 +625,13 @@ def test_handle_site(self, pghoard):
 
         # adjust basebackup interval to be slightly longer than what this
         # basebackup took and make sure it's not retriggered
-        site_config["basebackup_interval_hours"] = (time.monotonic() - backup_start + 1) / 3600
+        backup_duration = max((time.monotonic() - backup_start), 1)
+        site_config["basebackup_interval_hours"] = (backup_duration * 2) / 3600
         pghoard.handle_site(pghoard.test_site, site_config)
         assert pghoard.test_site not in pghoard.basebackups
 
         # create a new backup now that we have some state
-        time.sleep(2)
+        time.sleep(backup_duration * 2 + 1)
         pghoard.handle_site(pghoard.test_site, site_config)
         assert pghoard.test_site in pghoard.basebackups
```

Comment:
Can't this be put into a loop which does some *10 waiting period until it times out?

Reply:
I'm not sure what you mean here ?

Reply:
Our usual infra tests have a pattern like this:

```
timeout = now() + 30seconds 
while now() < timeout:
    # check condition
    sleep(1)
```

But I guess this is irrelevant if this isn't the line which is flaky.

Reply:
We will need to wait that period anyway, so I don't think it would save us much time.